### PR TITLE
Tyxml 4 anouncement.

### PR DESCRIPTION
**Pull request by Drup**
Please review for typos&co. :)

### CODE REVIEW

File: _posts/2016-05-20-tyxml4.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+title: TyXML 4.0.0
+layout: default
+author: The Ocsigen Team
+authorurl: https://ocsigen.org
+---
+
+This is with great pleasure that we are announcing the release of [TyXML 4.0.0][changelog]. This new version is focused on improving the user experience, both for old and new TyXML users.
```
Comment (gasche) [inline _posts/2016-05-20-tyxml4.md]: It is with great pleasure .. for both old and new TyXML users

File: _posts/2016-05-20-tyxml4.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+[changelog]: https://github.com/ocsigen/tyxml/releases/tag/4.0.0
+
+## New TyXML manual and improved documentation
+
+A new TyXML manual is now available [here][manual]. Documentation was also improved in the various TyXML modules. Do not hesitate to provide feedback on [our bug tracker][bugtracker]!
```
Comment (gasche) [inline _posts/2016-05-20-tyxml4.md]: improved _for_ the various TyXML modules?
Documentation of the various TyXML modules was also improved?

File: _posts/2016-05-20-tyxml4.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+let%html to_ocaml = "<a href='ocaml.org'>OCaml!</a>"
+{% endhighlight %}
+
+It supports insertion of ocaml code inside the HTML and can be used with all the TyXML modules (such as [`Tyxml_js`][] and [`Eliom_content`][]) and with SVG. A complete overview can be found [in the manual][ppx_manual].
+
+[`Tyxml_js`]: http://ocsigen.org/js_of_ocaml/2.7/api/Tyxml_js
+[`Eliom_content`]: http://ocsigen.org/eliom/5.0/api/client/Eliom_content.Html5
+[ppx_manual]: https://ocsigen.org/tyxml/4.0/manual/ppx
+
+This new PPX leverage the (awesome) [markup][] library and was contributed by [Anton Bachin](https://github.com/aantron).
```
Comment (gasche) [inline _posts/2016-05-20-tyxml4.md]: leverages?

File: _posts/2016-05-20-tyxml4.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+---
+title: TyXML 4.0.0
+layout: default
+author: The Ocsigen Team
+authorurl: https://ocsigen.org
+---
+
+This is with great pleasure that we are announcing the release of [TyXML 4.0.0][changelog]. This new version is focused on improving the user experience, both for old and new TyXML users.
```
Comment (vasilisp) [inline _posts/2016-05-20-tyxml4.md]: I would prefer:
- _It_ is with great pleasure
- _The_ new version
- for both old and new

File: _posts/2016-05-20-tyxml4.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+TyXML is a library for building statically correct HTML5 and SVG documents.
+It provides a set of combinators which use the OCaml type system to ensure the validity of the generated document. TyXML's combinators can be used to build textual HTML and SVG, but also DOM trees or reactive interfaces, using [Eliom][] and [Js_of_ocaml][].
+
+[Eliom]: https://ocsigen.org/eliom/manual/clientserver-html
+[Js_of_ocaml]: https://ocsigen.org/js_of_ocaml/api/Tyxml_js
+
+## New TyXML manual and improved documentation
+
+A new TyXML manual is now available [here][manual]. Documentation was also improved in the various TyXML modules. Do not hesitate to provide feedback on [our bug tracker][bugtracker]!
```
Comment (vasilisp) [inline _posts/2016-05-20-tyxml4.md]: The documentation of the various TyXML modules was also improved.

Feedback _via_ our bug tracker (sounds like you are asking feedback on the tracker itself).

File: _posts/2016-05-20-tyxml4.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+## HTML syntax with the new PPX
+
+It is now possible to use the standard HTML syntax:
+
+{% highlight ocaml %}
+open Tyxml
+let%html to_ocaml = "<a href='ocaml.org'>OCaml!</a>"
+{% endhighlight %}
+
+It supports insertion of ocaml code inside the HTML and can be used with all the TyXML modules (such as [Js_of_ocaml][] and [Eliom][]) and with SVG. A complete overview can be found [in the manual][ppx_manual].
```
Comment (vasilisp) [inline _posts/2016-05-20-tyxml4.md]: Capitalize OCaml.

File: _posts/2016-05-20-tyxml4.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+[ppx_manual]: https://ocsigen.org/tyxml/4.0/manual/ppx
+
+This new PPX leverage the (awesome) [markup][] library and was contributed by [Anton Bachin](https://github.com/aantron).
```
Comment (vasilisp) [inline _posts/2016-05-20-tyxml4.md]: I am not very comfortable with PPX used as a noun. PPX (syntax) extension?

File: _posts/2016-05-20-tyxml4.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+* Toplevel printers are now available for the base HTML and SVG implementations:
+{% highlight ocaml %}
+# Tyxml.Html.(div [pcdata "Oh!"]) ;;
+- : [> `Div ] Tyxml.Html.elt = <div>Oh!</div>
+{% endhighlight %}
+
+* The HTML and SVG combinators have received numerous improvements to make them more consistent and easier to use. This means that several modules and functions have been renamed and some types have been changed, which break compatibility with previous TyXML versions.
+
+* An healthy amount of new element and attributes.
```
Comment (vasilisp) [inline _posts/2016-05-20-tyxml4.md]: _A_ healthy (h is not silent like in French)

File: _posts/2016-05-20-tyxml4.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+* The full changelog is available [here][changelog].
+
+## Compatibility
+
+This new version breaks compatibility. Compatible versions of `js_of_ocaml` and Eliom are available in their respective `master` branches.
```
Comment (vasilisp) [inline _posts/2016-05-20-tyxml4.md]: breaks compatibility with ...

_the_ respective master branches

Comment (Drup) [inline _posts/2016-05-20-tyxml4.md]: Well, it breaks compat with everything. :3

**Comment by Drup** [comment]
Thanks @vasilisp ! I fixed most of them.

**Comment by aantron** [comment]
Looks good, except of course I would put the PPX before the manual, or mention it loudly in a brief sentence or clause in the first paragraph. But looks good :)

**Comment by vasilisp** [comment]
I agree on emphasizing the PPX work.

Also, maybe you want to wait a bit, given that a Friday evening announcement won't receive sufficient attention? Monday morning would be perfect.

**Comment by aantron** [comment]
Yes, the PPX is the new "shiny" in this release, IMO (of course the other work is important). But, should take advantage to generate interest :)

File: _posts/2016-05-20-tyxml4.md [hunk 10]
```diff
@@ -33,11 +33,11 @@ open Tyxml
 let%html to_ocaml = "<a href='ocaml.org'>OCaml!</a>"
 {% endhighlight %}
 
-It supports insertion of ocaml code inside the HTML and can be used with all the TyXML modules (such as [Js_of_ocaml][] and [Eliom][]) and with SVG. A complete overview can be found [in the manual][ppx_manual].
+It supports insertion of OCaml code inside the HTML and can be used with all the TyXML modules (such as [Js_of_ocaml][] and [Eliom][]) and with SVG. A complete overview can be found [in the manual][ppx_manual].
 
 [ppx_manual]: https://ocsigen.org/tyxml/4.0/manual/ppx
 
-This new PPX leverage the (awesome) [markup][] library and was contributed by [Anton Bachin](https://github.com/aantron).
+This new PPX syntax extension leverages the (awesome) [markup][] library and was contributed by [Anton Bachin](https://github.com/aantron).
```
Comment (aantron) [inline _posts/2016-05-20-tyxml4.md]: The library name is Markup.ml, `markup` is the name of the package.

**Comment by Drup** [comment]
Is that better ? :)

**Comment by Drup** [comment]
@vasilisp Agreed, I need to wait for merge in opam-repository anyway.

File: _posts/2016-05-20-tyxml4.md [hunk 11]
```diff
@@ -5,7 +5,7 @@ author: The Ocsigen Team
 authorurl: https://ocsigen.org
 ---
 
-It is with great pleasure that we are announcing the release of [TyXML 4.0.0][changelog]. The new version is focused on improving the user experience for both old and new TyXML users.
+It is with great pleasure that we are announcing the release of [TyXML 4.0.0][changelog]. The major features of this new release are a new syntax extension that allows to use the standard HTML syntax and an improved user experience for both old and new TyXML users.
```
Comment (vasilisp) [inline _posts/2016-05-20-tyxml4.md]: That allows _using standard HTML syntax_. I would rephrase the part about the user experience, given that it is not a "concrete" feature. Something like:

> The major new feature of this release is a new syntax extension that allows using standard HTML syntax. Various smaller improvements also contribute to an improved user experience, for old and new users alike.

File: _posts/2016-05-20-tyxml4.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+
+## Others
+
+* Toplevel printers are now available for the base HTML and SVG implementations:
+{% highlight ocaml %}
+# Tyxml.Html.(div [pcdata "Oh!"]) ;;
+- : [> `Div ] Tyxml.Html.elt = <div>Oh!</div>
+{% endhighlight %}
+
+* The HTML and SVG combinators have received numerous improvements to make them more consistent and easier to use. This means that several modules and functions have been renamed and some types have been changed, which break compatibility with previous TyXML versions.
```
Comment (aantron) [inline _posts/2016-05-20-tyxml4.md]: s/which break/which breaks/

File: _posts/2016-05-20-tyxml4.md [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+
+* A healthy amount of new element and attributes.
```
Comment (aantron) [inline _posts/2016-05-20-tyxml4.md]: s/element/elements/

File: _posts/2016-05-20-tyxml4.md [hunk 14]
```diff
@@ -0,0 +1,69 @@
+---
+title: TyXML 4.0.0
+layout: default
+author: The Ocsigen Team
+authorurl: https://ocsigen.org
+---
+
+It is with great pleasure that we are announcing the release of [TyXML 4.0.0][changelog]. The major features of this new release are a new syntax extension that allows to use the standard HTML syntax and an improved user experience for both old and new TyXML users.
```
Comment (aantron) [inline _posts/2016-05-20-tyxml4.md]: I would say "a new PPX syntax extension," – there already was `pa_tyxml`. Not sure how many people are aware of `pa_tyxml`, though.

**Comment by aantron** [comment]
Would still change the order of sections, but LGTM modulo nits.

**Comment by Drup** [comment]
All done! I'll squash/merge when the times come.

**Comment by vasilisp** [comment]
So it's official :). Well done @Drup @aantron!
